### PR TITLE
feat: Exclude Deleted Tables Flag (INT1-101)

### PR DESCRIPTION
**Pull request by Caleb-Hurshman**
## Changes
This PR adds the `--exclude-deleted-tables` flag, which ignores deleted flags when querying table storage metrics.
This will help larger environments improve data processing performance.

**Related Issue**: [#10637](https://github.com/grafana/support-escalations/issues/10637)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,21 +9,22 @@ Exports [Snowflake](www.snowflake.com) warehouse, database, table, and replicati
...
-      --account=ACCOUNT      The account to collect metrics for.
-      --username=USERNAME    The username for the user used when querying metrics.
-      --password=PASSWORD    The password for the user used when querying metrics.
-      --private-key-path     The path to the user's RSA private key file.
-      --private-key-password The password for the user's RSA private key.
-      --role="ACCOUNTADMIN"  The role to use when querying metrics.
-      --warehouse=WAREHOUSE  The warehouse to use when querying metrics.
-      --version              Show application version.
-      --log.level=info       Only log messages with the given severity or above. One of: [debug, info, warn, error]
-      --log.format=logfmt    Output format of log messages. One of: [logfmt, json]
```
Comment (StefanKurek) [inline README.md:26]: should there be a help message for exclude deleted?

File: cmd/snowflake-exporter/main.go [hunk 2]
```diff
@@ -42,6 +42,7 @@ var (
 	privateKeyPassword = kingpin.Flag("private-key-password", "The password for the user's RSA private key.").Envar("SNOWFLAKE_EXPORTER_PRIVATE_KEY_PASSWORD").String()
 	role               = kingpin.Flag("role", "The role to use when querying metrics.").Default("ACCOUNTADMIN").Envar("SNOWFLAKE_EXPORTER_ROLE").String()
 	warehouse          = kingpin.Flag("warehouse", "The warehouse to use when querying metrics.").Envar("SNOWFLAKE_EXPORTER_WAREHOUSE").Required().String()
+	excludeDeleted     = kingpin.Flag("exclude-deleted-tables", "Exclude deleted objects when collecting metrics.").Default("false").Bool()
```
Comment (StefanKurek) [inline cmd/snowflake-exporter/main.go:45]: you might want to specify specifically "table storage metrics" or something like that

**Review by StefanKurek — APPROVED** [review]
Overall looks good. A couple of minor things though

File: README.md (hunk 1 above)
Comment (Caleb-Hurshman) [reply README.md:26]: It's there, it's just not highlighted for some reason (line 24)

**Review by gaantunes — APPROVED** [review]
LGTM

**Pull request merged**